CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs

```diff
@@ -0,0 +1,241 @@
+                StartDate = DateTime.Now.Date,
+                FinishDate = DateTime.Now.AddMonths(3).Date,
+                StudentIds = new List<long>() { 1, 2, 3, 4 },
+                MentorIds = new List<long>() { 8, 9 }
+
+            };
+
+            var existingStudentGroup = new CreateStudentGroupDto()
+            {
+                Name = "Exists_test_name",
+                CourseId = 3,
+                StartDate = DateTime.Now.AddMonths(3).Date,
+                FinishDate = DateTime.Now.AddMonths(6).Date,
+                StudentIds = new List<long>() { 5, 6, 7, 8 },
+                MentorIds = new List<long>() { 10, 11 }
+
+            };
+
+            var withoutMentorsAndStudentsStudentGroup = new CreateStudentGroupDto()
+            {
+                Name = "Unique_test_name",
+                CourseId = 3,
+                StartDate = DateTime.Now.AddMonths(3).Date,
+                FinishDate = DateTime.Now.AddMonths(6).Date,
+                StudentIds = new List<long>(),
+                MentorIds = new List<long>()
+            };
+
+            var studentGroupRepositoryMock = new Mock<IStudentGroupRepository>();
+            var studentsRepositoryMock = new Mock<IStudentRepository>();
+            var mentorsRepositoryMock = new Mock<IMentorRepository>();
+
+            studentGroupRepositoryMock.Setup(x => x.Add(It.IsAny<StudentGroup>()));
+
+            studentsRepositoryMock.Setup(x => x.GetStudentsByIdsAsync(newStudentGroup.StudentIds))
+                .ReturnsAsync(new List<Student>()
+                {
+                    new Student { Id = 1 },
+                    new Student { Id = 2 },
+                    new Student { Id = 3 },
+                    new Student { Id = 4 }
+                });
+
+            mentorsRepositoryMock.Setup(x => x.GetMentorsByIdsAsync(newStudentGroup.MentorIds))
+                .ReturnsAsync(new List<Mentor>()
+                {
+                    new Mentor { Id = 18 },
+                    new Mentor { Id = 19 }
+                });
+
+            studentsRepositoryMock.Setup(x => x.GetStudentsByIdsAsync(withoutMentorsAndStudentsStudentGroup.StudentIds))
+               .ReturnsAsync(new List<Student>());
+
+            mentorsRepositoryMock.Setup(x => x.GetMentorsByIdsAsync(withoutMentorsAndStudentsStudentGroup.MentorIds))
+                .ReturnsAsync(new List<Mentor>());
+
+            studentGroupRepositoryMock.Setup(x => x.IsGroupNameExistAsync(newStudentGroup.Name))
+                    .ReturnsAsync(false);
+
+            studentGroupRepositoryMock.Setup(x => x.IsGroupNameExistAsync(existingStudentGroup.Name))
+                    .ReturnsAsync(true);
+
+            _unitOfWorkMock.Setup(x => x.StudentGroupRepository).Returns(studentGroupRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.StudentRepository).Returns(studentsRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(mentorsRepositoryMock.Object);
```

Comment:
you can divide it into several lines

Reply:
It is divided into same logical block, spaces can harm the logical structure of the code